test/tests/functional/pbs_test_tpp.py

```diff
@@ -825,6 +829,89 @@ def test_COMM_ROUTERS_with_nondefault_pbs_conf(self):
                           job=True, interactive=True, resv=True,
                           resv_job=True)
 
+    def calculate_no_of_wait_threads(self, comm_pid):
+        """
+        This function calculate the number of wait threads
+        :param comm_pid: Comm's Pid
+        :type comm_pid: Integer.
+        """
+        ps_cmd = "ps -eT | grep pbs_comm | wc -l"
+        rc = self.server.du.run_cmd(self.server.hostname,
+                                    cmd=ps_cmd, as_script=True)
+        num_wait_threads = int(rc['out'][0]) - 1
+        return num_wait_threads
+
+    def test_PBS_COMM_THREADS(self):
+        """
+        Test allowable values for PBS_COMM_THREADS
```

Comment:
are we actually testing allowable values? we assign values and check if that number of threads are getting created or not.